File: libs/grpc/clients/ytsr.go
package clients

import (
	"log"

	"github.com/twirapp/twir/libs/grpc/constants"
	"github.com/twirapp/twir/libs/grpc/ytsr"

	"google.golang.org/grpc"
)

func NewYtsr(env string) ytsr.YtsrClient {
	serverAddress := createClientAddr(env, "ytsr", constants.YTSR_SERVER_PORT)

	conn, err := grpc.Dial(
		serverAddress,
		defaultClientsOptions...,
	)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	c := ytsr.NewYtsrClient(conn)

	return c
}

File: libs/grpc/clients/ytsr.ts
import { ChannelCredentials, createChannel, createClient } from 'nice-grpc'

import { CLIENT_OPTIONS, createClientAddr, waitReady } from './helper'
import { PORTS } from '../constants/constants.js'
import { YtsrDefinition } from '../ytsr/ytsr'

import type { YtsrClient } from '../ytsr/ytsr'

export async function createYtsr(env: string): Promise<YtsrClient> {
	const channel = createChannel(
		createClientAddr(env, 'ytsr', PORTS.YTSR_SERVER_PORT),
		ChannelCredentials.createInsecure(),
		CLIENT_OPTIONS,
	)

	await waitReady(channel)

	const client = createClient(YtsrDefinition, channel)

	return client as any
}
